Pdf/PdfDocumentGenerator.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+using DocumentService.Pdf.Models;
+using System.Diagnostics;
+
+namespace DocumentService.Pdf
+{
+    public class FilePathDoesNotExist : Exception
+    {
+        public FilePathDoesNotExist(string msg) : base(msg)
+        {
+
+        }
+    }
+     
+    public class PdfDocumentGenerator
+    {
+        public static void generatePdfByTemplate(string templatePath, List<ContentMetaData> metaDataList, string outputFilePath)
+        {
+            try 
+            { 
+                if(!File.Exists(templatePath))
+                {
+                    throw new FilePathDoesNotExist("The file path you provided is not Valid"); 
+                }
+                
+                string modifiedHtmlFilePath =  ReplaceFileElementsWithMetaData(templatePath, metaDataList);
+                
+                ConversionCmd(modifiedHtmlFilePath, outputFilePath);
+
+            } 
+            catch(Exception e)
+            {
+                throw;
```

Comment:
Use the exception `e`. Throw the exception.

Reply:
e was unused. throwing it as exception in next line
catch(Exception e)
{
    throw e;
}